lib/lightning/workflows.ex

```diff
@@ -405,4 +405,8 @@ defmodule Lightning.Workflows do
   def jobs_ordered_subquery do
     from(j in Job, order_by: [asc: j.inserted_at])
   end
+
+  def has_newer_version?(%Workflow{lock_version: version}) do
+    from(s in Snapshot, where: s.lock_version > ^version) |> Repo.exists?()
+  end
```

Comment:
@elias-ba I'm committing a change here, this query will return true is _any_ snapshot exists with a greater version...

Which at least from my crude testing appears to have the potential to cause an infinite loop. If not that then a bunch of extra calls to reload the workflow only to get the same one back.

I'm a little confused on how to test this behaviour, I had to jam a `insert(:snapshot, lock_version: 10)` inside a test to find out that the extra calls, and if I make the function always return true then the tests take forever (assuming infinite loop here).

Basically, I think in a separate PR drop some comments on `maybe_switch_workflow_version` and `commit_latest_version` (or on slack..).

I can't quite work out exactly how this _should_ work,